File: backend/app/application/dto/criteria.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CriteriaGroupCreate(BaseModel):
    job_position_id: UUID
    label: str
    description: str | None = None
    sort_order: int = 0


class CriteriaGroupUpdate(BaseModel):
    label: str | None = None
    description: str | None = None
    sort_order: int | None = None


class CriteriaGroupResponse(BaseModel):
    id: UUID
    job_position_id: UUID
    label: str
    description: str | None
    sort_order: int
    created_at: datetime
    updated_at: datetime


class CriteriaItemCreate(BaseModel):
    criteria_group_id: UUID
    label: str
    description: str | None = None
    behavior_examples_text: str | None = None
    sort_order: int = 0
    is_active: bool = True


class CriteriaItemUpdate(BaseModel):
    label: str | None = None
    description: str | None = None
    behavior_examples_text: str | None = None
    sort_order: int | None = None
    is_active: bool | None = None


class CriteriaItemResponse(BaseModel):
    id: UUID
    criteria_group_id: UUID
    label: str
    description: str | None
    behavior_examples_text: str | None
    sort_order: int
    is_active: bool
    created_at: datetime
    updated_at: datetime


class CriteriaGroupWithItems(BaseModel):
    id: UUID
    job_position_id: UUID
    label: str
    description: str | None
    sort_order: int
    items: list[CriteriaItemResponse]

File: frontend/src/domain/entities/criteria.ts
export interface CriteriaGroup {
  id: string;
  job_position_id: string;
  label: string;
  description: string | null;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface CriteriaItem {
  id: string;
  criteria_group_id: string;
  label: string;
  description: string | null;
  behavior_examples_text: string | null;
  sort_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CriteriaGroupWithItems extends CriteriaGroup {
  items: CriteriaItem[];
}

export interface CriteriaGroupCreate {
  job_position_id: string;
  label: string;
  description?: string;
  sort_order?: number;
}

export interface CriteriaItemCreate {
  criteria_group_id: string;
  label: string;
  description?: string;
  behavior_examples_text?: string;
  sort_order?: number;
  is_active?: boolean;
}

